### ISSUE TITLE
Supports keys converted to atom type

### ISSUE THREAD

**Issue by Hentioe** [homanchou/elixir-xml-to-map#14]
It can look like this:

 `XmlToMap.naive_map(xml_text, keys: :atoms)` and `XmlToMap.naive_map(xml_text, keys: :atoms!)`

Refer to the related design of the Jason library: https://hexdocs.pm/jason/Jason.html#decode/2

**Comment by homanchou** [comment]
Well I guess I'm fine with that feature if there is a precedent for it.  I don't have time to code this up, but a PR with a test would be welcome!

**Comment by Hentioe** [comment]
OK I will open a PR when I have time!

**Comment by LeandroFigueredo** [comment]
@homanchou Using a internal xml in test file selftitled as `sample_xml`, can you confirm if the expected result is equals to the result bellow ? 

     
      %{
      orders: %{
        order: [
          %{
            billing_address: "My address",
            id: "123",
            items: %{
              item: [
                %{
                  "#content": %{
                    description: "Hat",
                    price: "5.99",
                    quantity: "1",
                    sku: %{"#content": "ABC", "-lang": "en"}
                  },
                  "-currency": "USD"
                },
                %{
                  "#content": %{
                    description: "Bat",
                    price: "9.99",
                    quantity: "2",
                    sku: "ABC"
                  },
                  "-currency": "USD"
                }
              ]
            }
          },
          %{
            billing_address: "Uncle's House",
            id: "124",
            items: %{
              item: %{
                "#content": %{
                  description: "Hat",
                  price: "5.99",
                  quantity: "2",
                  sku: "ABC"
                },
                "-currency": "USD"
              }
            }
          }
        ]
      }
    }

**Comment by homanchou** [comment]
Yes it looks correct by my eyeballing it.  But I'm confused why you are you asking me here when ... doesn't the very first test in the test file already assert something like this?

**Comment by LeandroFigueredo** [comment]
Thanks for your reply @homanchou . In fact, the structure of the first test returns exactly this value that I left as an example above, but I made all the keys to be converted to atom type. I built a helper module so that when I insert %{keys: :atoms} into the naive_map function, all keys can be returned in the requested format. I want to send a pull request with greater accuracy and I left the example above just to see if this is the expected return.
